### ISSUE TITLE
trying to update an AutomationRule Objec

### ISSUE THREAD

**Issue by JohnHib** [smartsheet/smartsheet-csharp-sdk#32]
Exception thrown: 'Smartsheet.Api.InvalidRequestException' in smartsheet-csharp-sdk.dll

An exception of type 'Smartsheet.Api.InvalidRequestException' occurred in smartsheet-csharp-sdk.dll but was not handled in user code

Required object attribute(s) are missing from your request: action.includedColumnIds.

in C# the Action object shows an **IncludeColumnIds** property but not an **IncludedColumnIds** one? where do I find the missing property is this a bug in the C# API code?

**Comment by JohnHib** [comment]
No movement on this issue?

**Comment by ggoranov-smar** [comment]
> Exception thrown: 'Smartsheet.Api.InvalidRequestException' in smartsheet-csharp-sdk.dll

@JohnHib can you provide code sample, logs and exception information for us to understand and investigate the first issue?

> in C# the Action object shows an IncludeColumnIds property but not an IncludedColumnIds one? where do I find the missing property is this a bug in the C# API code?

Regarding the second issue, we will investigate.

**Comment by JohnHib** [comment]
`                {
                    ScreenLogEntry = "Found sheet: " + strSSPName;
                    long lSheetId = (long)sPPSSheet.Id!;
                    int i = 0;
                    PaginatedResult<AutomationRule> AutomationRules = smartsheet.SheetResources.AutomationRuleResources.ListAutomationRules(lSheetId, null);
                    foreach (AutomationRule arPPS in AutomationRules.Data)
                    {
                        if ((bool)arPPS.Enabled!)
                        {
                            Debug.WriteLine("[" + i + "]    Id = '" + arPPS.Id + "'");
                            AutomationRule arPPS2 = smartsheet.SheetResources.AutomationRuleResources.GetAutomationRule(lSheetId, (long)arPPS.Id!);
                            AutomationAction aAction = arPPS2.Action;
                            string strActionType = aAction.Type.ToString()!;
                            switch (strActionType)
                            {
                                case "NOTIFICATION_ACTION":
                                    ScreenLogEntry = "Found '" + strActionType + "'";
                                    Debug.WriteLine("    Action Type .................. '" + aAction.Type + "'");
                                    Debug.WriteLine("    Action NotifhyAllSharedUsers . '" + aAction.NotifhyAllSharedUsers + "'");
                                    Debug.WriteLine("    Action Recipients ............ '" + aAction.Recipients + "'");
                                    Debug.WriteLine("    Action RecipientColumnIds .... '" + aAction.RecipientColumnIds + "'");
                                    Debug.WriteLine("    Action Type .................. '" + aAction.Type + "'");
                                    Debug.WriteLine("    Action Frequency ............. '" + aAction.Frequency + "'");
                                    Debug.WriteLine("    Action IncludeAllColumns ..... '" + aAction.IncludeAllColumns + "'");
                                    Debug.WriteLine("    Action IncludeColumnIds ...... '" + aAction.IncludeColumnIds + "'");
                                    Debug.WriteLine("    Action Subject ............... '" + aAction.Subject + "'");
                                    string strSubject = aAction.Subject;
                                    if (strSubject.StartsWith("New PO for:"))
                                    {
                                        ScreenLogEntry = "Found target Subject: '" + strSubject + "'";
                                        string strMessage = aAction.Message.Replace("Email you invoice", "Email your invoice");
                                        aAction.Message = strMessage;
                                        IList<System.Int64> longs = new List<System.Int64>(0);
                                        aAction.IncludeColumnIds = longs;
                                        ScreenLogEntry = "Action Message changed";
                                        Debug.WriteLine("    Action Message ...... '" + aAction.Message + "'");
                                        arPPS.Action = aAction;
                                        arPPS.CreatedAt = null;
                                        arPPS.CreatedBy = null;
                                        arPPS.ModifiedAt = null;
                                        arPPS.ModifiedBy = null;
                                        arPPS.UserCanModify = null;
                                        ScreenLogEntry = "Updating Automation Rule '" + arPPS.Id + "'";
                                        if (chkRunLive.Checked)
                                        {
                                            AutomationRule arRuleNew = smartsheet.SheetResources.AutomationRuleResources.UpdateAutomationRule(lSheetId, arPPS);     // !!! Exception thrown here
                                        }                                    
                                    }
                                    break;
                                default:
                                    Debug.WriteLine("-   Action Type ............ '" + aAction.Type + "'");
                                    Debug.WriteLine("-   Action Frequency ....... '" + aAction.Frequency + "'");
                                    Debug.WriteLine("-   Action Subject ......... '" + aAction.Subject + "'");
                                    break;
                            }
                            i++;
                        }
                    }
                }
`
!!! -> Exception thrown: 'Smartsheet.Api.InvalidRequestException' in smartsheet-csharp-sdk.dll

An exception of type 'Smartsheet.Api.InvalidRequestException' occurred in smartsheet-csharp-sdk.dll but was not handled in user code

Required object attribute(s) are missing from your request: action.includedColumnIds.

**Comment by ggoranov-smar** [comment]
Yeah, it looks like a typo. In Java and Python SDKs it is `includedColumnIds`:
* [Python](https://github.com/smartsheet/smartsheet-python-sdk/blob/a0f62c5269c302deb69a7e2cccbd2ca560ab8625/smartsheet/models/automation_action.py#L40)
* [Java](https://github.com/smartsheet/smartsheet-java-sdk/blob/76cdb1e803a2a5fa2a934c9359c777d15d58e1df/src/main/java/com/smartsheet/api/models/AutomationAction.java#L49C24-L49C41)

@roomote can you create a PR to fix this issue?
`includeColumnIds` has to be renamed to `includedColumnIds` in https://github.com/smartsheet/smartsheet-csharp-sdk/blob/d60b9ce3c2376dd5e096841a57c04963205c5590/smartsheet-csharp-sdk/main/Smartsheet/Api/Models/AutomationAction.cs#L49
